src/Symfony/Contracts/Service/Attribute/Autoconfigure.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\Service\Attribute;
+
+/**
+ * A tag to tell how a base type should be autoconfigured.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
+class Autoconfigure
```

Comment:
The attribute is not `final` on purpose: subclassing it would allow implementing `AutoTag\*` attributes as discussed in https://github.com/symfony/symfony/pull/39776#issuecomment-757836140

Reply:
i tend to be skeptical about sub-classing this attribute in general :)

mostly in terms of the extra keyword(s) that then need to be accounted for (eg. search in project) as DI config diverges

Reply:
I think there is no reason to close this. I also like the DX that would be provided by the ability to type `AutoTag\` and be suggested with all existing tags, no matter the leading namespace.

Reply:
i've changed my mind. All this is an implementation detail of `autoconfigure: true` from a consumer pov, and it should just work :)

the rest is then solved by https://github.com/symfony/symfony/issues/26295, which becomes more relevant btw ;)